newIDE/app/src/GameDashboard/LeaderboardAdmin/LeaderboardSortOptionsDialog.js

```diff
@@ -0,0 +1,220 @@
+// @flow
+import React from 'react';
+import { t, Trans } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import { type I18n as I18nType } from '@lingui/core';
+
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import { ColumnStackLayout } from '../../UI/Layout';
+import RaisedButton from '../../UI/RaisedButton';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+import Text from '../../UI/Text';
+import TextField from '../../UI/TextField';
+
+import { type LeaderboardSortOption } from '../../Utils/GDevelopServices/Play';
+import { Column, LargeSpacer, Line } from '../../UI/Grid';
+import HelpButton from '../../UI/HelpButton';
+import Checkbox from '../../UI/Checkbox';
+import { FormHelperText } from '@material-ui/core';
+import { MarkdownText } from '../../UI/MarkdownText';
+
+const isWholeNumber = (value: any): boolean =>
+  value === null || Number.isInteger(value);
+
+type SortOptions = {|
+  sort: LeaderboardSortOption,
+  extremeAllowedScore: ?number,
+|};
+
+type Props = {
+  open: boolean,
+  sort?: LeaderboardSortOption,
+  extremeAllowedScore?: number,
+  onSave: SortOptions => Promise<void>,
+  onClose: () => void,
+};
+
+const extremeAllowedScoreMax = Number.MAX_SAFE_INTEGER;
+const extremeAllowedScoreMin = Number.MIN_SAFE_INTEGER;
+
+function LeaderboardSortOptionsDialog({
+  open,
+  onClose,
+  onSave,
+  sort,
+  extremeAllowedScore,
+}: Props) {
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+
+  const [
+    extremeAllowedScoreError,
+    setExtremeAllowedScoreError,
+  ] = React.useState<?string>(null);
+  const [
+    displayExtremeAllowedScoreInput,
+    setDisplayExtremeAllowedScoreInput,
+  ] = React.useState<boolean>(extremeAllowedScore !== undefined);
+  const [
+    extremeAllowedScoreValue,
+    setExtremeAllowedScoreValue,
+  ] = React.useState<number>(extremeAllowedScore || 0);
+
+  const [sortOrder, setSortOrder] = React.useState<LeaderboardSortOption>(
+    sort || 'ASC'
+  );
+
+  const onSaveSettings = async (i18n: I18nType) => {
+    if (displayExtremeAllowedScoreInput) {
+      if (extremeAllowedScoreValue > extremeAllowedScoreMax) {
+        setExtremeAllowedScoreError(
+          i18n._(t`Extreme score must be lower than ${extremeAllowedScoreMax}.`)
+        );
+        return;
+      }
+      if (extremeAllowedScoreValue < extremeAllowedScoreMin) {
+        setExtremeAllowedScoreError(
+          i18n._(
+            t`Extreme score must be equal or higher than ${extremeAllowedScoreMin}.`
+          )
+        );
+        return;
+      }
+    }
+    if (!isWholeNumber(extremeAllowedScoreValue)) {
+      setExtremeAllowedScoreError(
+        i18n._(
+          t`The extreme score must be a whole number between ${extremeAllowedScoreMin} and ${extremeAllowedScoreMax}.`
+        )
+      );
+      return;
+    }
+    setIsLoading(true);
```

Comment:
It is never set to false, should it be somewhere? I am not sure since the parent closes the dialog if the api call fails